OpenRA.Game/Sound/Sound.cs

```diff
@@ -81,9 +83,20 @@ ISoundSource LoadSound(ISoundLoader[] loaders, IReadOnlyFileSystem fileSystem, s
 
 		public void Initialize(ISoundLoader[] loaders, IReadOnlyFileSystem fileSystem)
 		{
+			StopMusic();
+			soundEngine.ReleaseSourcePool();
+
+			if (sounds != null)
+				foreach (var soundSource in sounds.Values)
+					soundSource.Dispose();
+
+			GC.Collect();
+
 			sounds = new Cache<string, ISoundSource>(s => LoadSound(loaders, fileSystem, s));
 			currentSounds = new Dictionary<uint, ISound>();
+			getMusicSource = s => LoadSound(loaders, fileSystem, s);
```

Comment:
You could reuse this function if you liked:
```
getMusicSource = s => LoadSound(loaders, fileSystem, s);
sounds = new Cache<string, ISoundSource>(getMusicSource);
```

You may wish to use a more generic name such as `getSoundSource` if you do.